libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts

```diff
@@ -0,0 +1,293 @@
+      this.stateData = sData;
+      this.countyData = cData;
+      this.selectedDate = currentDateSelected;
+      mortalButtonSelected ? this.drawDeathMap() : this.drawCasesMap();
+    });
+  }
+
+  public drawCasesMap() {
+    const stateExpression = ['match', ['get', 'STATE']];
+    const countyExpression = ['match', ['get', 'fips']];
+
+    this.stateData[this.selectedDate].forEach(function(row) {
+      const number = row['infection_rate'];
+      const color =
+        number > 1000
+          ? '#AE8080'
+          : number > 500
+          ? '#D18080'
+          : number > 100
+          ? '#FC8B8B'
+          : number > 10
+          ? '#FDC4C4'
+          : number > 1
+          ? '#FDE6E6'
+          : '#FFFFFF';
+      stateExpression.push(row['STATE'], color);
+    });
+
+    this.countyData[this.selectedDate].forEach(function(row) {
+      const number = row['infection_rate'];
+      const color =
+        number > 1000
+          ? '#AE8080'
+          : number > 500
+          ? '#D18080'
+          : number > 100
+          ? '#FC8B8B'
+          : number > 10
+          ? '#FDC4C4'
+          : number > 1
+          ? '#FDE6E6'
+          : '#FFFFFF';
+      countyExpression.push(row['fips'], color);
+    });
+
+    stateExpression.push('rgba(255,255,255,1)');
+    countyExpression.push('rgba(255,255,255,1)');
+
+    this.mapService.loaded.subscribe((map) => {
+      map.setPaintProperty('covid-state', 'fill-color', stateExpression);
+      map.setPaintProperty('covid-county', 'fill-color', countyExpression);
+    });
+
+    //this.mapService.map.setPaintProperty('covid-state', 'fill-color', stateExpression);
+    //this.mapService.map.setPaintProperty('covid-county', 'fill-color', countyExpression);
+  }
+
+  public drawDeathMap() {
+    console.log('death');
+
+    const stateExpression = ['match', ['get', 'STATE']];
+    const countyExpression = ['match', ['get', 'fips']];
+
+    this.stateData[this.selectedDate].forEach(function(row) {
+      const number = row['death_rate'];
+      const color =
+        number > 100
+          ? '#54278f'
+          : number > 75
+          ? '#756bb1'
+          : number > 50
+          ? '#9e9ac8'
+          : number > 10
+          ? '#bcbddc'
+          : number > 1
+          ? '#dadaeb'
+          : '#f2f0f7';
+      stateExpression.push(row['STATE'], color);
+    });
+
+    this.countyData[this.selectedDate].forEach(function(row) {
+      const number = row['death_rate'];
+      const color =
+        number > 100
+          ? '#54278f'
+          : number > 75
+          ? '#756bb1'
+          : number > 50
+          ? '#9e9ac8'
+          : number > 10
+          ? '#bcbddc'
+          : number > 1
+          ? '#dadaeb'
+          : '#f2f0f7';
+      countyExpression.push(row['fips'], color);
+    });
+
+    stateExpression.push('rgba(255,255,255,1)');
+    countyExpression.push('rgba(255,255,255,1)');
+
+    this.mapService.loaded.subscribe((map) => {
```

Comment:
`loaded` is a long-lived observable. Schedule the subscription close on component destroy using the `takeUntil` operator

https://medium.com/angular-in-depth/the-best-way-to-unsubscribe-rxjs-observable-in-the-angular-applications-d8f9aa42f6a0

Or limit the number of emissions with the `take` rxjs operator.

https://www.learnrxjs.io/learn-rxjs/operators/filtering/take

Reply:
used `take(1)` wherever necessary throughout the code.